src/averageAges.js

```diff
@@ -1,64 +1,93 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
+
+const MALE = 'm';
+const FEMALE = 'f';
+
+function calculateAgeSum(people) {
+  return people.reduce((acc, person) => {
+    return acc + (person.died - person.born);
+  }, 0);
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredPeopleByMan = people.filter((person) => {
+    if (century) {
+      const personDeathCentury = Math.ceil(person.died / 100);
+
+      return century === personDeathCentury && person.sex === MALE;
+    }
+
+    return person.sex === MALE;
+  });
+
+  const ageSum = calculateAgeSum(filteredPeopleByMan);
+
+  return ageSum / filteredPeopleByMan.length;
```

Comment:
Dividing by the length could be transferred to the `calculateAgeSum` function and called `calculateAverageAge`.